tests/rptest/services/admin.py

```diff
@@ -844,6 +900,47 @@ def list_users(self, node=None, include_ephemeral: Optional[bool] = False):
         return self._request("get", "security/users", node=node,
                              params=params).json()
 
+    def list_user_roles(self):
+        return self._request("get", f"security/users/roles")
+
+    def create_role(self, role: str):
+        return self._request("post", "security/roles", json=dict(role=role))
+
+    def get_role(self, role: str):
+        return self._request("get", f"security/roles/{role}")
+
+    def delete_role(self, role: str):
+        return self._request("delete", f"security/roles/{role}")
+
+    def update_role(self, role: str, update: RoleUpdate):
+        return self._request("put",
+                             f"security/roles/{role}",
+                             json=update._asdict())
+
+    def list_roles(self,
+                   filter: Optional[str] = None,
+                   principal: Optional[str] = None):
+        param = "" if filter is None else f"?filter={filter}"
+        param += "" if principal is None else f"?principal={principal}"
+        return self._request("get", f"security/roles{param}")
```

Comment:
If we have both a filter and a principal then the url has two '?' symbols, eg. `?filter=ba?principal=alice` instead of `?filter=ba&principal=alice`. I don't know how the admin API would parse this or if it causes any problems, but I think it would be simpler to just use the standard library's `urllib.parse.urlencode`: https://stackoverflow.com/questions/15799696/how-to-build-urls-in-python-with-the-standard-library

Reply:
good call, yeah.

Reply:
done